### PR TITLE
Add many more tests

### PR DESCRIPTION
**Pull request by io7m**
## Description

This adds tests to test the performance of logins, annotations, searching, and feed navigation.

## Motivation and Context

We want to start performing load testing to see where to direct optimization efforts.

Affects: https://www.notion.so/lyrasis/Develop-performance-testing-script-69f511b345284e819781ba94c4821240

## How Has This Been Tested?

I ran the tests against a local CM.

## Checklist

- [x] I have updated the documentation accordingly.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by io7m** [comment]
I'm not clear on what's causing that error. That's not a dependency we seem to specify anywhere.

**Comment by jonathangreen** [comment]
I think once https://github.com/ThePalaceProject/circulation-load-test/pull/5 goes in, it should resolve the issues running the linting here.

File: README.md [hunk 1]
```diff
@@ -11,16 +13,57 @@ about writing tests.
...
-## Test details
+## Fair Warning
 
-The currently available load tests are documented inline in the
-[locustfile](locustfile.py).
+*Some tests involve borrowing books. Borrowing a book, on many CMs, will consume a license. This may cost money!*
+
+A good example is the [bookmark tests](src/circulation_load_test/common/cmsearchbookmark.py); in order to create
+bookmarks, the bookmarks must be associated with a loan. The bookmark test loans a random book, adds a lot of
+bookmarks, and then returns the book.
```
Comment (jonathangreen) [inline README.md:41]: It should be clear from the warning already here, but just to reiterate, maybe add a sentence warning that this should not be run against real production instances.

File: README.md [hunk 2]
```diff
@@ -11,16 +13,57 @@ about writing tests.
 
 ## Running tests
```
Comment (jonathangreen) [inline README.md]: Might be good to rename this to something, more clear like running load tests. And add a small development section that includes how to run the unit tests.

File: pyproject.toml [hunk 3]
```diff
@@ -6,16 +6,34 @@ requires = ["poetry-core"]
...
 readme = "README.md"
 version = "0"
 
 [tool.poetry.dependencies]
+atoma = ">=0.0.17,<0.0.18"
 locust = "^2.12.1"
 python = ">=3.7,<4"
 
 [tool.poetry.group.dev.dependencies]
 pre-commit = "^2.20.0"
```
Comment (jonathangreen) [inline pyproject.toml]: It looks like we need tox and pytest to run the tests, would be good to add this to the `dev.dependencies` section, and add tox to the `ci.dependencies` section. Something like this: 
https://github.com/ThePalaceProject/circulation/blob/dc7132ac8ebb6371317ed47a420694acdb7a19f9/pyproject.toml#L234-L265

File: tests/demo.py [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    user = CMTests(env)
+    document = CMLogin.login(user)
+    root = document.links[CMAuthenticationLinkType.CATALOG]
+    search_link = CMSearchAndBookmark.find_search_link(user, root)
+    search = CMSearchAndBookmark(search_link)
+    search.execute(user)
+
+
+if __name__ == "__main__":
+    main()
```
Comment (jonathangreen) [inline tests/demo.py:31]: How is this file intended to be used?

File: tox.ini [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+[testenv]
+deps =
+	pytest
+commands =
+    pytest
+
+[pytest]
+minversion = 7.2.2
+testpaths =
+    tests
```
Comment (jonathangreen) [inline tox.ini:13]: This can happen in a separate PR, if we want to land this one first, but since we have a couple tests, it would be nice to run these with GH actions.

**Review by jonathangreen — COMMENTED** [review]
This looks really great @io7m 🎉 

I made a couple minor comments. I'm fine with all the ones about CI being addressed in a follow up PR, if you want to land this one first, I was just making some notes as I went through the code.

Can you resolve the merge conflicts with my fix to linting on the `main` branch? I'm going to give this a test locally today, then we should be good to merge.

File: tests/demo.py (hunk 4 above)
Comment (io7m) [reply tests/demo.py:31]: Oh, this is just there so that I could test the various bits of code that talk to the CM without having to run the whole Locust setup. I can remove it, it's not needed.

Comment (io7m) [reply tests/demo.py:31]: Actually, I left it in there with a comment. It is useful during the development of new load tests. Saves someone five minutes of working out how to instantiate the classes that Locust normally instantiates.

**Review by jonathangreen — APPROVED** [review]
🚀
